src/commands/edge_app.rs

```diff
@@ -333,6 +335,29 @@ impl EdgeAppCommand {
         Ok(())
     }
 
+    fn ensure_secrets_defined(&self, app_id: &str) -> Result<(), CommandError> {
+        let installation_id = self.get_or_create_installation(app_id)?;
+
+        let undefined_secrets_response = commands::get(
+            &self.authentication,
+            &format!(
+                "v4/edge-apps/secrets/undefined?installation_id={}",
+                installation_id
+            ),
+        )?;
+
+        let titles = serde_json::from_value::<Vec<String>>(undefined_secrets_response)?;
```

Comment:
to be honest maybe returning the list of undefined secrets would be better? Looks more natural. You return it in error, but we return a lot of different types of errors.
Returning the list of undefined secrets and checking that its length is 0 sounds better to me. What do you think?

Reply:
@sergey-borovkov what you mean is that it should not be an error, but just a list response?